### PR TITLE
TILA-1743: Add reservation user filter

### PR DESCRIPTION
**Pull request by saaste**
This also addresses TILA-1491 by moving permission checks behind filter.

### CODE REVIEW

File: api/graphql/reservations/reservation_filtersets.py [hunk 1]
```diff
@@ -22,6 +29,12 @@ class ReservationFilterSet(django_filters.FilterSet):
     # Filter for displaying reservations which requires or had required handling.
     requested = django_filters.BooleanFilter(method="get_requested")
 
+    only_with_permission = django_filters.BooleanFilter(
+        method="get_only_with_permission"
+    )
+
+    user = django_filters.NumberFilter(method="get_user")
```
Comment (nicobav) [inline api/graphql/reservations/reservation_filtersets.py:36]: We could use `ModelChoiceFilter` and then we don't need the get_user  method

File: api/graphql/tests/test_reservations/test_reservation_queries.py [hunk 2]
```diff
@@ -296,6 +340,149 @@ def test_reservation_total_count(self):
         assert_that(content.get("errors")).is_none()
         self.assertMatchSnapshot(content)
 
+    def test_hide_fields_with_personal_information(self):
+        self.create_reservation_by_admin()
+        self.client.force_login(self.regular_joe)
+        response = self.query(
```
Comment (nicobav) [inline api/graphql/tests/test_reservations/test_reservation_queries.py:346]: Lots of duplicate query string, perhaps add a class method for getting it?
Also could add begin and end fields at least to confirm that reservation time things are visible.

**Review by nicobav — CHANGES_REQUESTED** [review]
Looks nice :+1: Only some friendly suggestions to ponder about

File: api/graphql/reservations/reservation_filtersets.py (hunk 1 above)
Comment (saaste) [reply api/graphql/reservations/reservation_filtersets.py]: Oh, true! Replaced 🧹

File: api/graphql/tests/test_reservations/test_reservation_queries.py (hunk 2 above)
Comment (saaste) [reply api/graphql/tests/test_reservations/test_reservation_queries.py:382]: Great ideas! Implemented both 🧹

**Review by nicobav — APPROVED** [review]
Magnificent :shipit:

**Pull request merged**